app/configurator/configurator-state.tsx

```diff
@@ -1107,30 +1109,26 @@ const reducer: Reducer<ConfiguratorState, ConfiguratorStateAction> = (
 
     case "CHART_CONFIG_UPDATE_COLOR_MAPPING":
       if (draft.state === "CONFIGURING_CHART") {
-        const { dimensionIri, values, random } = action.value;
-        if (
-          isSegmentColorMappingInConfig(draft.chartConfig) &&
-          draft.chartConfig.fields.segment &&
-          draft.chartConfig.fields.segment.componentIri === dimensionIri
-        ) {
+        const { field, dimensionIri, values, random } = action.value;
+        const fieldValue = get(draft.chartConfig, `fields.${field}`) as
+          | GenericSegmentField
+          | CategoricalColorField
+          | undefined;
```

Comment:
Can we try not to have this assertion ? If we had types or segment field later, this assertion will be false.

Reply:
Would a `isColorMappingInConfig` helper function work in this case (to check if there is a segment or color field with type of categorical)?

Reply:
Here we want to have a any type of field with a palette right ? What could be good would be to have a type `type ColorField = Extract<ChartFields, { palette: any }>` (example code, not working) ?